### PR TITLE
Improve performance of BigInteger.Log

### PR DESCRIPTION
**Pull request by axelheer**
Since only the most significant bits of a big number are relevant for computing its logarithm (the current implementation multiplies with 0.5 after processing a bit, which naturally doesn't have any effect after 54 iterations), using the build-in LOG function just for the highest 64-bit block and adjusting the result considering the actual bit length should be sufficient.

#1307

### CODE REVIEW

**Comment by axelheer** [comment]
cc: @stephentoub @KrzysztofCwalina

File: src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs [hunk 1]
```diff
@@ -990,26 +990,20 @@ public static Double Log(BigInteger value, Double baseValue)
...
-                }
-                indbit = 0x80000000;
-            }
-            return (Math.Log(c) + log2 * bitlen) / Math.Log(baseValue);
+            // extract most significant bits
+            ulong x = (h << 32 + c) | (m << c) | (l >> 32 - c);
+
+            // let v = value, b = bit count, x = v/2^b-64
+            // log ( v/2^b-64 * 2^b-64 ) = log ( x ) + log ( 2^b-64 )
+            return Math.Log(x, baseValue) + Math.Log(Math.Pow(2, b - 64), baseValue);
```
Comment (axelheer) [inline src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs]: Looking at it again, we could replace `Math.Log(Math.Pow(2, b - 64), baseValue)` with `(b - 64) / Math.Log(baseValue, 2)`. Which is the nicer solution?

**Comment by stephentoub** [comment]
Thanks, @axelheer. "should be sufficient"?  Otherwise, LGTM.

**Comment by axelheer** [comment]
Is sufficient, of course. Forgive my English, at my place people like using subjunctive forms too much. ;)

**Comment by stephentoub** [comment]
Ok, thanks, just verifying :smile:

**Pull request merged**